src/Database/IdentifierQuoter.php

```diff
@@ -106,7 +109,11 @@ public function quoteExpression(ExpressionInterface $expression): void
     protected function _quoteParts(Query $query): void
     {
         foreach (['distinct', 'select', 'from', 'group'] as $part) {
-            $contents = $query->clause($part);
+            try {
+                $contents = $query->clause($part);
+            } catch (InvalidArgumentException) {
```

Comment:
What throws this now?

Reply:
The method is used by all query types and not all queries have the parts which are being checked after the split. `Query::clause()` throws an exception for non existent clause. I can change the parts list based on query type if this try/catch is undesirable.

Reply:
At best, this needs to be well-documented.